src/ensembl/utils/database/unittestdb.py

```diff
@@ -138,3 +169,20 @@ def _load_data(self, conn: sqlalchemy.engine.Connection, table: str, src: StrPat
             conn.execute(text(f"BULK INSERT {table} FROM '{src}'"))
         else:
             conn.execute(text(f"LOAD DATA LOCAL INFILE '{src}' INTO TABLE {table}"))
+
+
+class UnitTestDBContext:
+    """Context manager that returns a `UnitTestDB` when using `with`, and drops it afterwards."""
+
+    def __init__(self, *args, **kwargs) -> None:
+        self.args = args
+        self.kwargs = kwargs
+        self.test_db: UnitTestDB | None = None
+
+    def __enter__(self):
+        self.test_db = UnitTestDB(*self.args, **self.kwargs)
+        return self.test_db
+
+    def __exit__(self, *args):
+        if self.test_db:
+            self.test_db.drop()
```

Comment:
I remember reading and writing a comment about this but cannot find it :shrug: So basically you can add the `__enter__` and `__exit__` method to `UnitTestDB` and make this object both available via standard constructor or context manager. I think it may be best for simplicity.

Reply:
That sounds like a better idea, I'll try

Reply:
Done, seems to work fine